asset-transfer-basic/application-gateway-java/src/main/java/application/java/App.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Running TestApp: 
+// gradle runApp 
+
+package application.java;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.PrivateKey;
+import org.hyperledger.fabric.client.identity.Signer;
+import java.security.cert.CertificateException;
+import java.util.concurrent.TimeUnit;
+import org.hyperledger.fabric.client.*;
+import org.hyperledger.fabric.client.CallOption;
+import org.hyperledger.fabric.client.CommitException;
+import org.hyperledger.fabric.client.identity.*;
+import org.hyperledger.fabric.client.Contract;
+import org.hyperledger.fabric.client.Gateway;
+import org.hyperledger.fabric.client.identity.Identities;
+import org.hyperledger.fabric.client.Network;
+import java.security.cert.X509Certificate;
+import java.time.LocalDateTime;
+
+import io.grpc.ManagedChannel;
+import io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder;
+import io.grpc.protobuf.StatusProto;
+import java.security.InvalidKeyException;
+import org.hyperledger.fabric.client.identity.Signers;
+import org.hyperledger.fabric.client.GatewayException;
+import org.hyperledger.fabric.client.SubmittedTransaction;
+
+
+public class App {
+
+        private static final String mspID     = "Org1MSP";
+        private static final String channelName = "mychannel";
+        private static final String chaincodeName = "basic";
+        private static final Path cryptoPath  = Paths.get("..","..", "test-network", "organizations", "peerOrganizations", "org1.example.com");
+        private static final Path certPath    = cryptoPath.resolve(Paths.get("users", "User1@org1.example.com", "msp", "signcerts", "cert.pem"));
+        private static final Path keyPath     = cryptoPath.resolve(Paths.get("users", "User1@org1.example.com", "msp", "keystore"));
+        private static final Path tlsCertPath = cryptoPath.resolve(Paths.get("peers", "peer0.org1.example.com", "tls", "ca.crt"));
+
+        
+	public static void main(String[] args) throws CommitException, InterruptedException, IOException, CertificateException,InvalidKeyException {
```

Comment:
I don't think CommitException can be thrown here, as the transaction invocation all happen within a try/catch block.